source/manuals/programming-languages/nodejs/style-guide.html.md.erb

```diff
@@ -0,0 +1,122 @@
+---
+title: GDS Node.js Style Guide
+---
+
+# <%= current_page.data.title %>
+
+## General style
+
+As with all JavaScript at GDS, for Node.js applications we use the [standard] style.
+
+## Use strict mode
+
+[Strict mode] should be enabled for Node.js modules, as in the browser this prevents polluting the global scope (in
+Node.js this would pollute the `global` object, rather than the `window` object).
+
+## ES2015+
+
+Because with Node.js you control the version of the interpreter you're code is running on,
+you have access to many the latest features of JavaScript, such as:
+
+- [let] and [const]
+- [Arrow functions]
+- [Array.prototype] methods like `Array.map()`, `Array.find()` and `Array.values()`
+- The [spread operator]
+
+However use caution when using cutting edge JavaScript features and be sure to read the [documentation].
+
+A full list of what features are supported in each version of Node.js can be found at [Node Green].
+
+
+### let and const
+
+For Node.js code, be explicit about variable mutability using [const] and [let]:
+
+```
+const foo // GOOD: explicitly immutable variable declaration
```

Comment:
Since const does not actually guarantee immutability, I've seen people avoid it entirely, what do you think?

Reply:
Is it still worth using to convey the meaning that this variable shouldn't be changed to a developer? For example if I see an array or an object set used with `const` I would avoid changing the inner values (even though I can). I know another pattern that tends to be used is uppercase names for `const`.

Reply:
I'd also add that in the case of const objects, it's more about intention than reality, as we all know all this blocks is re-assignment.

Might also be worth highlighting the upsides/downsides of let/const in terms of memory use. With const you rely on the garbage collector eventually freeing up memory, so their use is contextual.

Reply:
@nickcolley I still generally find it helps keep intention clear using `const` over `var` or `let`. But I agree that if we are going to use `const` the style guide should include a reminder (and link to MDN) to how it only guarantees referential immutability for complex types. 

@brid In reality, my general feeling is that if we need to manually reassign variables to a `null` value when we've finished with them in order to help the GC; we're probably doing something wrong elsewhere in the code. I think we should perhaps include a section in the style guide on avoiding unnecessary variable declarations. Helping developers to think about writing performant code as they go along, rather than having to optimise things.